src/governance/VoltVetoGovernor.sol

```diff
@@ -0,0 +1,398 @@
+    function updateTimelock(
+        address newTimelock
+    ) external onlyVoltRole(VoltRoles.GOVERNOR) {
+        _updateTimelock(newTimelock);
+    }
+
+    function _updateTimelock(address newTimelock) private {
+        emit TimelockChange(address(_timelock), newTimelock);
+        _timelock = TimelockController(payable(newTimelock));
+    }
+
+    /// ------------------------------------------------------------------------
+    /// Vote counting
+    /// ------------------------------------------------------------------------
+
+    /**
+     * @dev Supported vote types. Matches Governor Bravo ordering.
+     */
+    enum VoteType {
+        Against,
+        For,
+        Abstain
+    }
+
+    struct ProposalVote {
+        uint256 againstVotes;
+        uint256 forVotes;
+        uint256 abstainVotes;
+        mapping(address => bool) hasVoted;
+    }
+
+    mapping(uint256 => ProposalVote) private _proposalVotes;
+
+    /**
+     * @dev See {IGovernor-COUNTING_MODE}.
+     */
+    // solhint-disable-next-line func-name-mixedcase
+    function COUNTING_MODE()
+        public
+        pure
+        virtual
+        override
+        returns (string memory)
+    {
+        return "support=bravo&quorum=against";
+    }
+
+    /**
+     * @dev See {IGovernor-hasVoted}.
+     */
+    function hasVoted(
+        uint256 proposalId,
+        address account
+    ) public view virtual override returns (bool) {
+        return _proposalVotes[proposalId].hasVoted[account];
+    }
+
+    /**
+     * @dev Accessor to the internal vote counts.
+     */
+    function proposalVotes(
+        uint256 proposalId
+    )
+        public
+        view
+        virtual
+        returns (uint256 againstVotes, uint256 forVotes, uint256 abstainVotes)
+    {
+        ProposalVote storage proposalvote = _proposalVotes[proposalId];
+        return (
+            proposalvote.againstVotes,
+            proposalvote.forVotes,
+            proposalvote.abstainVotes
+        );
+    }
+
+    /**
+     * @dev See {Governor-_quorumReached}.
+     */
+    function _quorumReached(
+        uint256 proposalId
+    ) internal view virtual override returns (bool) {
+        ProposalVote storage proposalvote = _proposalVotes[proposalId];
+
+        return
+            quorum(proposalSnapshot(proposalId)) <= proposalvote.againstVotes;
+    }
+
+    /**
+     * @dev See {Governor-_voteSucceeded}. In this module, the forVotes must be strictly over the againstVotes.
+     * @dev Veto votes are always considered "successful" in this part of the logic, as there is no opposition
+     * between 'for' and 'against' votes, since people cannot vote 'for'. For a veto to be considered successful,
+     * it only needs to reach quorum.
+     */
+    function _voteSucceeded(
+        uint256 proposalId
+    ) internal view virtual override returns (bool) {
+        ProposalVote storage proposalvote = _proposalVotes[proposalId];
+
+        return proposalvote.againstVotes != 0;
```

Comment:
I would make this entire function a no-op as the `state` function does not need it, and the overridden `state` function only needs to check `_quorumReached` to know if a vote has passed.

Reply:
good idea